nari/ext/act/utils.py

```diff
@@ -11,34 +11,29 @@
 DEFAULT_DATE_FORMAT: str = '%Y-%m-%dT%H:%M:%S.%f%z'
 
 
-# pylint: disable=invalid-name
-class ActLogChecksumType(Enum):
-    """List of hashsum algorithms used by different ACT versions"""
-    MD5 = "md5"
-    SHA256 = "sha256"
-# pylint: enable=invalid-name
+try:
+    from nari_act_rust.utils import date_from_cs_string, validate_checksum_internal
+except ImportError:
+    def date_from_cs_string(datestr: str) -> Timestamp:
+        """Look, this is dirty. This is wrong. Please someone find a better way to do this."""
+        return int(datetime.strptime(f'{datestr[:26]}{datestr[-6:]}', DEFAULT_DATE_FORMAT).timestamp() * 1000)
 
+    def validate_checksum_internal(line: str, index: int) -> bool:
+        """Validates an ACT log line internal function"""
+        parts = line.split('|')
+        check_hash = parts[-1]
+        to_hash = f'{"|".join(parts[:-1])}|{index}'.encode('utf-8')
```

Comment:
May be bette to do these independently. In the off chance only one fails, at least you get the other. Deferring to @nonowazu's opinion tho.

Reply:
Oh yeah those should be done aswell